common/cpp/crypto/crypto_utils.cpp

```diff
@@ -52,27 +44,28 @@ namespace Error = tcf::error;
 
 /**
  * Generate a cryptographically strong random bit string.
+ * Implemented with getrandom() GRND_NONBLOCK.
  * Throws: RuntimeError.
  */
 ByteArray pcrypto::RandomBitString(size_t length) {
     char err[constants::ERR_BUF_LEN];
     ByteArray buf(length);
-    int res = 0;
+    ssize_t   res = 0;
 
     if (length < 1) {
         std::string msg("Crypto Error (RandomBitString): "
             "length argument must be at least 1");
         throw Error::ValueError(msg);
     }
 
-    res = RAND_bytes(buf.data(), length);
-
-    if (res != 1) {
-        std::string msg("Crypto Error (RandomBitString): ");
-        ERR_load_crypto_strings();
-        ERR_error_string(ERR_get_error(), err);
-        msg += err;
-        throw Error::RuntimeError(msg);
+    // Get random bytes until buffer is full
+    for (ssize_t i = length; i > 0; i =- res) {
+        res = getrandom(&buf.data()[length - i], i, GRND_NONBLOCK);
```

Comment:
Avalon crypto library is used by Enclave. So systems calls cannot be used. 
https://github.com/hyperledger/avalon/blob/6039768e7314ed1eea5472923559daa2799f061e/common/cpp/crypto/CMakeLists.txt#L52